pkg/authenticator/ecrsecret.go

```diff
@@ -0,0 +1,110 @@
+package authenticator
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"os"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	varPackagesNamespace = "eksa-packages"
+	varConfigMapName     = "installed-namespaces"
+	varConfigMapKey      = "NAMESPACES"
+	varECRTokenName      = "ecr-token"
+)
+
+type ecrSecret struct {
+	clientset kubernetes.Interface
+}
+
+var _ Authenticator = (*ecrSecret)(nil)
+
+func NewECRSecret(config *rest.Config) *ecrSecret {
+	clientset, _ := kubernetes.NewForConfig(config)
+
+	return &ecrSecret{
+		clientset: clientset,
+	}
+}
+
+func (s *ecrSecret) AuthFilename() (string, error) {
+	// Check if Helm registry config is set
+	helmconfig := os.Getenv("HELM_REGISTRY_CONFIG")
+	if helmconfig != "" {
+		// Use HELM_REGISTRY_CONFIG
+		return helmconfig, nil
+	}
+
+	return "", nil
```

Comment:
Shouldn't we return an error not nil? Otherwise remove the error from the return types.

Reply:
Removed